build.gradle

```diff
@@ -216,9 +216,39 @@ task distNativeSources(type: Zip) {
     }
 }
 
+task distEndorsedCommonSources(type: Zip) {
+    dependsOn ":endorsedLibraries:endorsedLibsCommonSources"
+
+    destinationDirectory = file("$distDir/sources")
+    archiveFileName = "kotlin-endorsed-common-sources.zip"
+
+    includeEmptyDirs = false
+    include('**/*.kt')
+
+    from(project(':endorsedLibraries').file('build/sources-common'))
+}
+
+task distEndorsedNativeSources(type: Zip) {
+    dependsOn ":endorsedLibraries:endorsedLibsNativeSources"
+
+    destinationDirectory = file("$distDir/sources")
+    archiveFileName = "kotlin-endorsed-native-sources.zip"
```

Comment:
No, no. Please put the source code of every single library in a separate ZIP file. Please use pattern `LIBRARY-sources.zip` for the ZIP file name.